tests/integration/cmd_describe_component_test.go

```diff
@@ -181,79 +223,230 @@ var _ = Describe("odo describe component command tests", func() {
+									} else {
+										helper.JsonPathDoesNotExist(stdout, "devForwardedPorts.0.platform")
+										helper.JsonPathDoesNotExist(stdout, "runningOn")
+									}
+								}
+							})
+
+							By("running with default output", func() {
+								cmd := helper.Cmd("odo", "describe", "component")
+								if experimental {
+									cmd = cmd.AddEnv("ODO_EXPERIMENTAL_MODE=true")
+								}
+								res := cmd.ShouldPass()
+								stdout := res.Out()
+								checkDevfileDescription(stdout, false)
+								if podman {
+									if experimental {
+										Expect(stdout).To(ContainSubstring("Forwarded ports"))
+										Expect(stdout).To(ContainSubstring("[podman] 127.0.0.1:" + ports["3000"][len("127.0.0.1:"):] + " -> runtime:3000"))
+										Expect(stdout).NotTo(ContainSubstring("[cluster] 127.0.0.1:"))
+										Expect(stdout).To(ContainSubstring("Running on:"))
+										Expect(stdout).To(ContainSubstring("podman: Dev"))
+										Expect(stdout).NotTo(ContainSubstring("cluster: "))
+									} else {
+										Expect(stdout).To(ContainSubstring("Running in: None"))
+										Expect(stdout).NotTo(ContainSubstring("Forwarded ports"))
+										Expect(stdout).NotTo(ContainSubstring("127.0.0.1:"))
+										Expect(stdout).NotTo(ContainSubstring("Running on:"))
+										Expect(stdout).NotTo(ContainSubstring("podman: "))
+										Expect(stdout).NotTo(ContainSubstring("cluster: "))
+									}
+								} else {
+									Expect(stdout).To(ContainSubstring("Forwarded ports"))
+									if experimental {
+										Expect(stdout).To(ContainSubstring("[cluster] 127.0.0.1:" + ports["3000"][len("127.0.0.1:"):] + " -> runtime:3000"))
+										Expect(stdout).NotTo(ContainSubstring("[podman] 127.0.0.1:"))
+										Expect(stdout).To(ContainSubstring("Running on:"))
+										Expect(stdout).To(ContainSubstring("cluster: Dev"))
+										Expect(stdout).NotTo(ContainSubstring("podman: "))
+									} else {
+										Expect(stdout).To(ContainSubstring("127.0.0.1:" + ports["3000"][len("127.0.0.1:"):] + " -> runtime:3000"))
+										Expect(stdout).NotTo(ContainSubstring("[cluster] 127.0.0.1:"))
+										Expect(stdout).NotTo(ContainSubstring("[podman] 127.0.0.1:"))
+										Expect(stdout).NotTo(ContainSubstring("Running on:"))
+										Expect(stdout).NotTo(ContainSubstring("podman: "))
+										Expect(stdout).NotTo(ContainSubstring("cluster: "))
+									}
+								}
+							})
+						})
+
+					It(fmt.Sprintf("should describe the component from another directory (experimental=%s)", strconv.FormatBool(experimental)),
+						func() {
+							By("running with json output", func() {
+								err := os.Chdir("/")
+								Expect(err).NotTo(HaveOccurred())
+								cmd := helper.Cmd("odo", "describe", "component", "--name", cmpName, "-o", "json")
+								if experimental {
+									cmd = cmd.AddEnv("ODO_EXPERIMENTAL_MODE=true")
+								}
+								var stdout string
+								var stderr string
+								if experimental || !podman {
```

Comment:
This series of conditions is not very clear. Especially the else's. Do you think we could find another way to express them?

Reply:
I think we can explicitly write them as separate `It` specs. Even if it might make repeating some parts, I think it should make it easier to read and analyze. What do you think?

Reply:
Yes, it would be better I think

Reply:
Sorry for the delay - I've rewritten the tests as separate specs when possible. It should be easier to read now, I think. Please take a look at [`806bdab` (#6431)](https://github.com/redhat-developer/odo/pull/6431/commits/806bdab58b4b7c1322bdabbcf578c1d041fc38ae)

Reply:
I find this version more understandable. I think it will also be easier to cleanup when we decide to remove the experimental mode.

Thanks!

/lgtm